src/merkle_reg.rs

```diff
@@ -122,6 +122,21 @@ impl<T> MerkleReg<T> {
         self.dag.get(&hash).or_else(|| self.orphans.get(&hash))
     }
 
+    /// Returns the parents of a node
+    pub fn parents(&self, hash: Hash) -> Content<T> {
+        let nodes = self.dag.get(&hash).map(|node| {
+            node.parents
+                .iter()
+                .copied()
+                .filter_map(|leaf| self.dag.get(&leaf).map(|node| (leaf, node)))
+                .collect()
+        });
+
+        Content {
+            nodes: nodes.unwrap_or(BTreeMap::new()),
+        }
+    }
```

Comment:
What do you think of moving this to into Content? We'd introduce a new ContentNode type for dealing with the contents of any single node, e.g. take a look at this sketch:

```rust
struct ContentNode<'a, T> {
    reg: &'a MerkleReg<T>,
    node: &'a Node<T>,
}

impl<'a> ContentNode<'a, T> {
    fn parents(&self) -> Content<T> {
        Content {
            reg: self.reg,
            nodes: self.node.parents.filter_map(|p| self.reg.node(p)).collect()
        }
    }

    fn value(&self) -> &T {
        &self.node.value
    }
}

struct Content<'a, T> {
   reg: &'a MerkleReg<T>
   nodes: BTreeMap<Hash, ContentNode<'a, T>>
}

impl Content {
    fn nodes() -> Vec<ContentNode<'a, T>> {
        self.nodes.values().collect()
    }
}
```
Then for this dag, the flow would be:
```
a  b
|\ |
| \|
c  d
 \/
  e
```
```rust
let content = reg.read().
content.values() // --> {a, b}
content.nodes() // -> [ContentNode(a_node), ContentNode(b_node) ]
content.nodes()[0].value() // -> a
content.nodes()[1].value() // -> b
content.nodes()[0].parents() // -> Content<T> at the time `a` was written
content.nodes()[0].parents().values() // -> { c, d }
content.nodes()[1].parents().values() // -> { d }
content.nodes()[0].parents().nodes()[0].value() // -> c
content.nodes()[0].parents().nodes()[0].parents() // -> Content at the time`c` was written
content.nodes()[0].parents().nodes()[0].parents().values() // -> {e}
content.nodes()[0].parents().nodes()[0].parents().nodes()[0].value() // -> e
content.nodes()[0].parents().nodes()[0].parents().nodes()[0].parents().values() // -> {}
```

Reply:
ok, looks neat, I'll add that now and push a new commit